aces/idt/core/common.py

```diff
@@ -756,3 +758,153 @@ def format_exposure_key(key: str) -> str:
+    return threshold
+
+
+def find_close_indices(data: np.array, threshold: float = 0.005) -> list:
+    """
+    Find the indices of rows that have any values with differences below the threshold.
+
+    Parameters
+    ----------
+    data : np.array
+        A numpy array of shape (n, 3, 3) where each row contains RGB values.
+    threshold : float, optional
+        The tolerance threshold for determining if the RGB values between consecutive
+        rows are similar. Default is 0.005.
+
+    Returns
+    -------
+    List[int]
+        A list of indices where the rows have RGB differences below the threshold.
+    """
+    top_indices = []
+    bottom_indices = []
+    n_rows = data.shape[0]
+
+    # Search from the top
+    for i in range(n_rows - 1):
+        if np.any(np.abs(data[i] - data[i + 1]) < threshold):
+            top_indices.append(i)
+        else:
+            break
+
+    # Search from the bottom
+    for i in range(n_rows - 1, 0, -1):
+        if np.any(np.abs(data[i] - data[i - 1]) < threshold):
+            bottom_indices.append(i)
+        else:
+            break
+
+    # Combine and return the indices
+    return sorted(np.array(top_indices + bottom_indices))
+
+
+class ExposureClippingFilter:
+    """
+    The ExposureClippingFilter class is used to filter out exposure values that have
+    clipped RGB values. The class compares the RGB values of each image sample in
+    consecutive exposures. If any of the RGB values are less than the threshold, the
+    exposure is considered clipped and should be removed.
+    """
+
+    def __init__(self, data_dict: dict, threshold: float):
+        """
+        Initialize the ExposureClippingFilter with a data dictionary and a threshold
+        value.
+
+        Parameters
+        ----------
+        data_dict : dict
+            Dictionary with exposure values as keys and numpy arrays of image samples
+            as values.
+        threshold : float
+            The threshold for the comparison.
+        """
+        self.data_dict = data_dict
+        self.threshold = threshold
+
+    def compare_image_samples(self, samples1: np.ndarray, samples2: np.ndarray) -> bool:
+        """
+        Compare the r, g, and b values of each corresponding image sample in two arrays.
+        If any of the r, g, b values for any of the images are less than the threshold,
+        return True. Otherwise, return False.
+
+        Parameters
+        ----------
+        samples1 : np.ndarray
+            First set of image samples with shape (n, 3).
+        samples2 : np.ndarray
+            Second set of image samples with shape (n, 3).
+
+        Returns
+        -------
+        bool
+            True if any r, g, b values are less than the threshold, False otherwise.
+        """
+        if not isinstance(samples1, np.ndarray):
+            samples1 = np.array(samples1)
+
+        if not isinstance(samples2, np.ndarray):
+            samples2 = np.array(samples2)
```

Comment:
It is better to use `np.asarray` here without testing, if the array is already a `ndarray` it is passed as is thus this should be: `np.abs(np.asarray(samples1) - np.asarray(samples2))`

Reply:
Even better would to use `colour.utilities.as_float_array`, the benefit is that this allows to set the processing to a smaller bit-depth globally.

Reply:
It seems a bit overkill to have a method here for what is really a single line:

`np.any(np.abs(np.asarray(samples1) - np.asarray(samples2)) < self.threshold))`

It could actually even be rewritten as follows:

`np.any(np.subtract(samples1, samples2) < self.threshold))`

Reply:
Yeah this came from the unit testing using lists vs the application using numpy arrays, we can loose the typing check all together and then loose the function